src/main/java/com/microsoft/sqlserver/jdbc/SQLServerParameterMetaData.java

```diff
@@ -703,58 +703,66 @@ public boolean isWrapperFor(Class<?> iface) throws SQLException {
         return t;
     }
 
-    private Map<String, Object> getParameterInfo(int param) throws SQLServerException {
-        boolean paramFound = false;
+    private Map<String, Object> getParameterInfo(int param) {
         if ((stmtParent).bReturnValueSyntax && isTVP) {
-            paramFound = procMetadata.size() >= param;
-            if (paramFound) {
-                return procMetadata.get(param - 1);
-            }
+            return procMetadata.get(param - 1);
         } else {
             // Note row 1 is the 'return value' meta data
-            paramFound = procMetadata.size() > param;
-            if (paramFound) {
-                return procMetadata.get(param);
-            }
+            return procMetadata.get(param);
```

Comment:
do we not want to check and error if procMetadata.size() < param anymore?

Reply:
`checkParam()` takes care of that - it wouldn't reach here in that case.